New sub-section and standard names for land surface variables

SamanthaPullen opened the pull request:
This PR addresses Issue #59 

This PR adds some variable names to the CCPP Standard Names for land surface variables, in preparation for Met 
Office work on land surface DA with JEDI. I think it makes sense to add a new sub-section for these surface variables. I have added a subsection `land_surface`, but it could be a wider scope sub-section covering all surface (i.e. land, sea, sea-ice). Please advise :)

I have added the following land surface variable names, in alphabetical order (which would be nice to see elsewhere in the CCPP Standard Names list, for ease of reference):

1. `land_ice_area_fraction`
2. `mass_content_of_water_in_top_soil_layer`
3. `surface_snow_density`
4. `urban_area_fraction`
5. `volume_fraction_of_liquid_water_in_soil_at_critical_point`
6. `volume_fraction_of_liquid_water_in_soil_at_saturation`
7. `volume_fraction_of_liquid_water_in_soil_at_wilting_point`

I have used [CF Standard Names (cfconventions.org)](https://cfconventions.org/Data/cf-standard-names/current/build/cf-standard-name-table.html) where possible, and where nothing suitable exists I have based the proposed CCPP name on the CCPP or CF name of a related or similar variable.

Opened as a draft PR for now to allow coordination between Met Office and JCSDA before submitting for review. Comments welcome in the meantime!

ClaraDraper-NOAA commented:
> These additions have been discussed with @ClaraDraper-NOAA and she has confirmed that she is happy with them.

My apologies, I responded only to the email. These names are fine by me. Thanks for getting this ready @SamanthaPullen

nusbaume commented on standard_names.xml:
```diff
@@ -418,6 +418,35 @@
+                   long_name="fraction of horizontal area of grid cell that is urban">
+      <type kind="kind_phys" units="frac">real</type>
+    </standard_name>
+    <standard_name name="volume_fraction_of_liquid_water_in_soil_at_critical_point"
```
I noticed that along with these names there appears to be GFS/UFS variables with names like `volume_fraction_of_soil_moisture` and `volumetric_soil_moisture`.  Would these quantities be considered the same as, say, `volume_fraction_of_liquid_water`?  If so then we should probably come to an agreement of which phrasing is best (or if both should be allowed).

Pinging @climbfuji @grantfirl @dudhia @mkavulich

ClaraDraper-NOAA replied:
Note that the GFS models (Noah, Noah-MP) have two categories of prognostic soil moisture variable: 
(total) volumetric soil moisture  - this means liquid plus solid 
liquid volumetric soil moisture - liquid component only

The ones you reference above are probably the totals (and I'd support making sure the CCPP standard names include "total"). Note, other models might track this differently, and instead use liquid and solid as their prognostic variables.

nusbaume replied:
Thanks for the clarification @ClaraDraper-NOAA!  It looks like there is a rule already that states that the phrase "soil moisture" means all phases of water, so apologies for not noticing that earlier.

Also it looks like there is another rule that says `volume_fraction` should be used instead of `volumetric`.  That won't impact this PR, but it does mean that we'll need to make an issue to replace the standard names that are using `volumetric` to instead use `volume_fraction` (which is really more of a note to myself).

nusbaume reviewed (changes requested):
Looks good, but I did have one concern that should probably be discussed more widely amongst the group, or at least among folks who know more about land surface modeling than me.

nusbaume reviewed (approved):
My one concern has been resolved.  Thanks!

SamanthaPullen commented:
There is some associated discussion on the naming conventions and precise definitions of variables for area fractions in #57. This could have implications for the variable names `land_ice_area_fraction` and `urban_area_fraction` in this PR

SamanthaPullen commented:
I have switched this PR to draft given the related discussion in #57. Will update names for `land_ice_area_fraction` and `urban_area_fraction` by appending `_of_cell_area` after the Easter Holiday and re-open for review

SamanthaPullen commented:
@mkavulich @ss421 @svahl991 Following discussion at #57 and for consistency with the naming convention for sea ice fraction variables proposed there, I have modified the following land surface variable names:

- `land_ice_area_fraction`
- `urban_area_fraction`

modified to:

-  `land_ice_area_fraction_of_cell_area`
- `urban_area_fraction_of_cell_area`

Now ready for review again - @nusbaume are you happy with this update?

MayeulDestouches reviewed (approved):
Thanks @SamanthaPullen for this addition. I approve these new names.

nusbaume commented:
@SamanthaPullen Thanks for double-checking with me!  Yes those changes look fine on my end, so my approval still holds.

SamanthaPullen commented:
@nusbaume , @mkavulich, could this PR be merged please? It has had 2 approvals, but strangely only 1 of them is showing in the panel below

climbfuji commented:
@SamanthaPullen I'll ping folks directly to make sure this PR gets reviewed promptly. Sorry for the delay.

climbfuji reviewed (approved):
I am in favor of adding these new names. It would be good if @mkavulich and/or @grantfirl also reviewed the PR.

mkavulich reviewed (approved):
@SamanthaPullen thank you for making the discussed changes, sorry for the delay. Merging now.

SamanthaPullen commented:
@mkavulich thank you:)

grantfirl commented:
Late to the party, but looks fine to me.
